compiler/qsc_partial_eval/tests/intrinsics.rs

```diff
@@ -1054,3 +1054,39 @@ fn call_to_pauli_i_rotation_for_global_phase_is_noop() {
             Return"#]],
     );
 }
+
+#[test]
+fn call_to_operation_with_codegen_intrinsic_override_should_skip_impl() {
+    let program = get_rir_program(indoc! {"
+        namespace Test {
+            operation Op1() : Unit {
+                body intrinsic;
+            }
+            @SimulatableIntrinsic()
+            operation Op2() : Unit {
+                Op1();
+            }
+            operation Op3() : Unit {
+                Op1();
+            }
+            @EntryPoint()
+            operation Main() : Unit {
+                Op1();
+                Op2();
+                Op3();
+            }
+        }
+    "});
+
```

Comment:
It would be good to use `assert_callable` like in other tests in this file to verify that the call IDs match with the correct callables.

Reply:
Added!